### DISCUSSION TITLE
ubuntu Error : update-alternatives: --install needs <link> <name> <path> <priority>

### DISCUSSION THREAD

**Discussion by Sir-Osama** [CloudBytes-Academy/CloudBytes.dev#59] [Q&A] [answered]
I'm tryin to update python to the latest version  I installed python3.9 and I need to add the old version and the new version to 'alternatives' so typed :
`sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8`
and I get this error : 
`update-alternatives: --install needs <link> <name> <path> <priority>`
What should I do ??

**Comment by rehanhaider** [comment, marked as answer]
Hey @Sir-Osama , notice the 1 & 2 at the end of the lines. you need to keep them while executing these. 

```
sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.6 1
sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.9 2
```

After that run the below and choose the selection that you want as default.
```
sudo update-alternatives --config python3
```

Let me know if this solves your issue and please star the repository if this was helpful 🙂

**Reply by Sir-Osama** [reply]
this exactly what i did the only difference that I typed pyhon3.8 instade of python3.6 which is my current version of python on my device 
and it steal give me the same Error :
`update-alternatives: --install needs <link> <name> <path> <priority>`

**Reply by rehanhaider** [reply]
Can you share screenshots please. Of the command you entered and the error message.

Also, share the output of
```
python --version
```

**Reply by Sir-Osama** [reply]
Sure ,  : 
![Screenshot from 2021-10-11 17-31-22](https://user-images.githubusercontent.com/86108333/136817911-78dde96d-6aa0-4261-be3b-a68ef0657358.png)

**Comment by rehanhaider** [comment]
Your command is wrong.

Where is the  "1" at the end?. Look at the example I provided above, it has a 1 at the end of the command that signifies the priority.

**Reply by Sir-Osama** [reply]
Oh sorry I thought it's only for clarification :) 
and thank you sir for helping me , now I'm going to complete the process .
many thanks!

**Reply by rehanhaider** [reply]
I'm glad it helped. Star the repo please :)

**Reply by Sir-Osama** [reply]
Yeah I already did, but now i had a new problem actually this one is a big problem do I nead to open a new discussion for it?

**Comment by rehanhaider** [comment]
Either is fine. But new thread is better for future reference

**Reply by K2AAE** [reply]
Thanks for the guide and the 1,2 pointer. Just wish linux/python would figure out the update issues, I shouldn't have to stop and spend hours tracking down how to upgrade.

**Reply by rehanhaider** [reply]
True. Official repository takes almost a year to update the version, even today v3.10 is not available by default.
